designs/sam_destroy_cmd.md

```diff
@@ -0,0 +1,180 @@
+SAM Destroy Command
+====================
+
+This is the design for a command to **destroy** a CloudFormation stack.
+
+What is the problem?
+--------------------
+
+Customers create CloudFormation stacks using `sam deploy`. After deploying the stacks, customers may have to delete
+a stack to cleanup all of their resources and stacks because of stacks that landed in ROLLBACK_FAILED, 
+rapid testing of development tasks, and removing defective code that may have a large security risk.
+
+Currently, Customers would have to open up the CloudFormation console every time they have to delete, forcing them to leave the cli or use the `aws cloudformation` command. 
+This worsens developer experience as it requires them to leave their sam workflow to find what they are looking for. 
+
+ 
+What will be changed?
+---------------------
+
+In this proposal, we will be providing a new command, `sam destroy`, to
+destroy CloudFormation stacks in the cloud. 
+Customers will be able to view failures of the commands and other events directly inside the cli.
+
+
+Success criteria for the change
+-------------------------------
+- Customers will be able to delete a stack in the cloud
+
+Out-of-Scope
+------------
+- Deleting and cleaning up resources that cannot be deleted or are in DELETE_FAILED state
+
+User Experience Walkthrough
+---------------------------
+Customers first uses sam deploy to deploy the SAM/CloudFormation stack to the cloud. 
+Once the customer has finished using the stack, the customer will run `sam destroy --stack-name <STACK_NAME>` 
+to destroy the stack.
+
+Implementation
+==============
+The stack will be deleted with `cfn.delete_stack(StackName=stack_name, **args)`. 
+
+To handle errors better, the code will first check if there is a valid aws account, the stack specified exists, and 
+if any resources are in the state DELETE_FAILED they are specified in the retained_resources section.
+
+The two other errors that are specially handled are TerminationProtection and AccessDeniedException.
+
+TerminationProtection will be checked if termination protection is turned on and provide the following error message.
+
+```sh
+The stack {stack_name} has TerminationProtection turned on. Disable it on the aws console at 
+https://us-west-1.console.aws.amazon.com/cloudformation/home \n or run aws 
+cloudformation update-termination-protection --stack-name {stack_name} 
+--no-enable-termination-protection 
+```
+
+AccessDeniedException will be checked if the user does not have the access to delete the CloudFormation stack.
+```sh
+The user account does not have access to delete the stack. Add the cloudformation:delete policy 
+with the following format to the user account.
+{ 
+    "Version": "2012-10-17", 
+    "Statement": [ 
+        { 
+            "Effect": "Allow", 
+            "Action": [ 
+            "cloudformation:delete" 
+            ], 
+            "Resource": "*" 
```

Comment:
Lets just add the corresponding resource instead of all the resources. also does the rolearn specified require permissions to delete other resources within the stack? say sqs, sns, apigateway? how do we handle that?